### PR TITLE
feat(pkg/apis/config) add the protocol to the mesh cluster

### PR DESCRIPTION
**Pull request by steeling**
Add the port protocol to the service mesh.

Signed-off-by: Sean Teeling <seanteeling@microsoft.com>

Please mark with X for applicable areas.

-->
**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ x] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
No

1. Is this a breaking change?
No

### CODE REVIEW

File: charts/osm/crds/multiclusterservice.yaml [hunk 1]
```diff
@@ -40,9 +40,13 @@ spec:
               required:
                 - serviceAccount
                 - clusters
+                -protocol
               properties:
                 serviceAccount:
                   type: string
+                protocol:
+                  description: The IP protocol for this port. Supports "TCP", "UDP", and "SCTP". Default is TCP.
```
Comment (snehachhabria) [inline charts/osm/crds/multiclusterservice.yaml:48]: If you want a fixed set of allowed values please follow this example : https://github.com/openservicemesh/osm/blob/f9dc7b9be2287303345154d5b490763d3a051e3e/charts/osm/crds/meshconfig.yaml#L54

Further please define a default as well in the CRD

Comment (snehachhabria) [reply charts/osm/crds/multiclusterservice.yaml]: This hasn't yet been addressed for the protocol

File: charts/osm/crds/multiclusterservice.yaml [hunk 2]
```diff
@@ -43,6 +43,22 @@ spec:
...
+                  type: array
+                  items:
+                    type: object
+                    required:
+                      - port
+                      - protocol
+                    properties:
+                      port: 
+                        description: The port that will be exposed by this service.
+                        type: integer
```
Comment (snehachhabria) [inline charts/osm/crds/multiclusterservice.yaml:57]: please add a minimum and maximum value for the post like this : https://github.com/openservicemesh/osm/blob/bc441bae296c306f37515387f8a92d8f38660657/charts/osm/crds/policy.yaml#L89

Comment (steeling) [reply charts/osm/crds/multiclusterservice.yaml:57]: done!

File: charts/osm/crds/multiclusterservice.yaml (hunk 1 above)
Comment (steeling) [reply charts/osm/crds/multiclusterservice.yaml]: done!

**Review by shashankram — APPROVED** [review]

**Review by snehachhabria — APPROVED** [review]

**Pull request merged**